src/oop/object_interactions/medium/q2/MainTest.java

```diff
@@ -0,0 +1,224 @@
+        assertTrue(testClass.hasModifier(writeInDiaryMethodName, null, "public"), String.format(incorrectModifierMessage, writeInDiaryMethodName, className));
+        assertTrue(testClass.hasReturnType(writeInDiaryMethodName, null, Void.TYPE), String.format(incorrectReturnType, writeInDiaryMethodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("hearSecretInputProvider")
+    @Tag("dependent1")
+    public void correctWriteInDiaryMethod(String name, String secret, String[] allFriends, String friend, String friendSecret) throws Throwable {
+        Object[][] arguments = {
+                {name, String.class},
+                {secret, String.class},
+                {allFriends, String[].class}
+        };
+        assertTrue(testClass.hasMethod(writeInDiaryMethodName, null),
+                "Your " + writeInDiaryMethodName + " method does not have the correct name of arguments.");
+        assertTrue(testClass.hasMethod(writeInDiaryMethodName, null, Void.TYPE),
+                "Your " + writeInDiaryMethodName + " method does not have the correct return type.");
+        assertTrue(testClass.hasMethod(writeInDiaryMethodName, null, Void.TYPE, new String[]{"public"}),
+                "Your " + writeInDiaryMethodName + " method does not have the correct modifier.");
+        Object classInstance = testClass.createInstance(arguments);
+        String[] friendNames = (String[]) testClass.getFieldValue(classInstance, "friendNames");
+        String[] friendSecrets = (String[]) testClass.getFieldValue(classInstance, "friendSecrets");
+        int friendIndex = friendSecrets.length;
+        for (int i = 0; i < friendNames.length; i++)
+            if (friendNames[i].equals(friend)) {
+                friendIndex = i;
+                break;
+            }
+        if (friendIndex < friendNames.length)
+            friendSecrets[friendIndex] = friendSecret;
+
+        boolean flg = false;
+        for (int i = 0; i < allFriends.length; i++) {
+            if (friend.equals(allFriends[i]))
+                flg = true;
+        }
+        testClass.setFieldValue(classInstance, friendSecrets, "friendSecrets");
+        if (flg) {
+            testClass.callMethod(writeInDiaryMethodName, null, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("I have the following secrets:"),
+                    new Optional(new StringLiteral(" ")),
+                    new NewLine(),
+                    new StringLiteral(friend + " told me " + friendSecret),
+                    new Optional(new StringLiteral(".")),
+                    new Optional(new StringLiteral(" "))
+            });
+        } else {
+            testClass.callMethod(writeInDiaryMethodName, null, new String[]{"public"}, classInstance, new Clause[]{
+                    new StringLiteral("I have the following secrets:"),
+                    new Optional(new StringLiteral(" "))
+            });
+        }
+
+    }
+
```

Comment:
Please add a test which uses both the methods together to make sure they integrate correctly.